BuildResidentialScheduleFile/tests/build_residential_schedule_file_test.rb

```diff
@@ -0,0 +1,459 @@
+    assert_in_epsilon(8760, sf.annual_equivalent_full_load_hrs(col_name: 'plug_loads_vehicle', schedules: sf.tmp_schedules), 0.1)
+    assert_in_epsilon(3671, sf.annual_equivalent_full_load_hrs(col_name: 'plug_loads_well_pump', schedules: sf.tmp_schedules), 0.1)
+    assert_in_epsilon(2074, sf.annual_equivalent_full_load_hrs(col_name: 'fuel_loads_grill', schedules: sf.tmp_schedules), 0.1)
+    assert_in_epsilon(3671, sf.annual_equivalent_full_load_hrs(col_name: 'fuel_loads_lighting', schedules: sf.tmp_schedules), 0.1)
+    assert_in_epsilon(3671, sf.annual_equivalent_full_load_hrs(col_name: 'fuel_loads_fireplace', schedules: sf.tmp_schedules), 0.1)
+    assert_in_epsilon(2471, sf.annual_equivalent_full_load_hrs(col_name: 'pool_pump', schedules: sf.tmp_schedules), 0.1)
+    assert_in_epsilon(2471, sf.annual_equivalent_full_load_hrs(col_name: 'pool_heater', schedules: sf.tmp_schedules), 0.1)
+    assert_in_epsilon(2502, sf.annual_equivalent_full_load_hrs(col_name: 'hot_tub_pump', schedules: sf.tmp_schedules), 0.1)
+    assert_in_epsilon(2650, sf.annual_equivalent_full_load_hrs(col_name: 'hot_tub_heater', schedules: sf.tmp_schedules), 0.1)
+    assert_in_epsilon(2994, sf.annual_equivalent_full_load_hrs(col_name: 'hot_water_dishwasher', schedules: sf.tmp_schedules), 0.1)
+    assert_in_epsilon(4158, sf.annual_equivalent_full_load_hrs(col_name: 'hot_water_clothes_washer', schedules: sf.tmp_schedules), 0.1)
+    assert_in_epsilon(4204, sf.annual_equivalent_full_load_hrs(col_name: 'hot_water_fixtures', schedules: sf.tmp_schedules), 0.1)
+    assert(!sf.schedules.keys.include?('vacancy'))
+  end
+
+  def test_smooth_vacancy
+    hpxml = _create_hpxml('base.xml')
+    XMLHelper.write_file(hpxml.to_oga, @tmp_hpxml_path)
+
+    @args_hash['schedules_vacancy_period'] = 'Dec 1 - Jan 31'
+    @args_hash['output_csv_path'] = File.absolute_path(File.join(@tmp_output_path, 'smooth-vacancy.csv'))
+    model, hpxml, result = _test_measure()
+
+    info_msgs = result.info.map { |x| x.logMessage }
+    assert(info_msgs.any? { |info_msg| info_msg.include?('smooth schedule') })
+    assert(info_msgs.any? { |info_msg| info_msg.include?('CalendarYear=2007') })
+    assert(info_msgs.any? { |info_msg| info_msg.include?('NumberOfTimestepsPerHour=1') })
+    assert(info_msgs.any? { |info_msg| info_msg.include?('State=CO') })
+    assert(!info_msgs.any? { |info_msg| info_msg.include?('RandomSeed') })
+    assert(info_msgs.any? { |info_msg| info_msg.include?('GeometryNumOccupants=3.0') })
+    assert(info_msgs.any? { |info_msg| info_msg.include?('VacancyPeriod=Dec 1 - Jan 31') })
+
+    sf = SchedulesFile.new(model: model, schedules_path: @args_hash['output_csv_path'], col_names: Constants.ScheduleColNames.keys)
+
+    vacancy_hrs = 31 * 2 * 24
+    occupied_ratio = (1 - vacancy_hrs / 8760)
+
+    assert_in_epsilon(4997 * occupied_ratio, sf.annual_equivalent_full_load_hrs(col_name: 'occupants', schedules: sf.tmp_schedules), 0.1)
```

Comment:
I'm probably just not thinking straight, but why is the 4997 number here different than the 6020 number from `test_smooth`? I thought you get the same schedules in the CSV either way, except for a vacancy column that will now have 1s in it.

Reply:
Good catch. Because, e.g., `1488/8760 = 0` in ruby.